build.sbt

```diff
@@ -135,9 +135,9 @@ lazy val r2dbc =
     .settings(headerSettings(IntegrationTest))
     .settings(Defaults.itSettings)
     .settings(Dependencies.r2dbc)
+    .settings(Scala3.settings)
```

Comment:
when I tried I noticed some compiler warnings, we just ignore them or what is the approach?

Reply:
I will review them, thanks for pointing that out.

Reply:
There is one scala3 compile [warning left](https://github.com/akka/akka-projection/blob/main/akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala#L169-L172) which we should keep, but I am not able to suppress it with `@nowarn` because in scala2 this is not a warn.

Reply:
We can accept some warnings in the Scala 3 compile but good to fix those that are possible.

That one you linked is interesting. It's a sealed class in akka and it's kind of a guard in case some additional implementation is added. Strange that Scala 3 is so strict about a sealed class in a dependency.